src/mbgl/tile/geometry_tile_data.cpp

```diff
@@ -17,94 +17,38 @@ static double signedArea(const GeometryCoordinates& ring) {
     return sum;
 }
 
-static LinearRing<std::int32_t> toWagyuPath(const GeometryCoordinates& ring) {
-    LinearRing<std::int32_t> result;
+static LinearRing<int32_t> toWagyuPath(const GeometryCoordinates& ring) {
+    LinearRing<int32_t> result;
     result.reserve(ring.size());
     for (const auto& p : ring) {
         result.emplace_back(p.x, p.y);
     }
     return result;
 }
 
-static void pushWagyuRing(GeometryCollection & solution,
-                          mapbox::geometry::wagyu::ring_ptr<std::int32_t> r) {
-    GeometryCoordinates lr;
-    lr.reserve(r->size() + 1);
-    auto firstPt = r->points;
-    auto ptIt = r->points;
-    do {
-        lr.emplace_back(ptIt->x, ptIt->y);
-        ptIt = ptIt->prev;
-    } while (ptIt != firstPt);
-    lr.emplace_back(firstPt->x, firstPt->y); // close the ring
-    solution.push_back(lr);
-}
-
-static void buildWagyuResults(GeometryCollection & solution,
-                              mapbox::geometry::wagyu::ring_vector<std::int32_t>& rings) {
-    for (auto r : rings) {
-        if (r == nullptr) {
-            continue;
-        }
-        assert(r->points);
-        if (r->size() < 3) {
-            continue;
-        }
-        solution.emplace_back();
-        pushWagyuRing(solution, r);
-        for (auto c : r->children) {
-            if (c == nullptr) {
-                continue;
-            }
-            assert(c->points);
-            if (c->size() < 3) {
-                continue;
-            }
-            pushWagyuRing(solution, c);
-        }
-        for (auto c : r->children) {
-            if (c == nullptr) {
-                continue;
-            }
-            if (!c->children.empty()) {
-                buildWagyuResults(solution, c->children);
-            }
+static GeometryCollection toGeometryCollection(MultiPolygon<int16_t>&& multipolygon) {
+    GeometryCollection result;
```

Comment:
I did, but because `result` is a flattened set of rings, it would require a pre-iteration over `multipolygon` in order to calculate the size, so I went with the simpler code.